connectivity/drivers/emac/TARGET_STM/stm32xx_emac.cpp

```diff
@@ -960,7 +962,17 @@ void STM32_EMAC::remove_multicast_group(const uint8_t *addr)
 
 void STM32_EMAC::set_all_multicast(bool all)
 {
-    /* No-op at this stage */
+    STM32_EMAC &emac = STM32_EMAC::get_instance();
+
+    ETH_MACFilterConfigTypeDef pFilterConfig;
+    
+    if (HAL_ETH_GetMACFilterConfig(&emac.EthHandle, &pFilterConfig) != HAL_OK){
+        return;
+    }
+
+    pFilterConfig.PassAllMulticast = all ? ENABLE : DISABLE;
```

Comment:
This isn't valid, given the way you've implemented `add_multicast_group`. The sequence.

    add_multicast_group(X);
    set_all_multicast(true);
    set_all_multicast(false);

would leave you not receiving X. If you're aiming to keep it as simple as possible while being valid, make `set_all_multicast(true)` turn on `PassAllMulticast`, and have `false` do nothing.

(But if you were implementing groups properly, your code would be the correct and canonical implementation for the "all multicast" flag.)